wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -335,6 +361,34 @@ private static EnvironmentImpl copyEnv(Workspace workspace, String envName) {
         return new EnvironmentImpl(environment);
     }
 
+    private class CleanupRuntimeOnAbnormalRuntimeStop implements EventSubscriber<RuntimeStatusEvent> {
+        @Override
+        public void onEvent(RuntimeStatusEvent event) {
+            if (event.isFailed()) {
+                RuntimeState state = runtimes.remove(event.getIdentity().getWorkspaceId());
+                if (state != null) {
+                    eventService.publish(DtoFactory.newDto(WorkspaceStatusEvent.class)
+                                                   .withWorkspaceId(state.runtime.getContext().getIdentity()
+                                                                         .getWorkspaceId())
+                                                   .withPrevStatus(WorkspaceStatus.STOPPING)
```

Comment:
Are you sure it's _STOPPING_ ?
It looks like if the stop is abnormal then the flow should be _RUNNING -> STOPPED_